src/cmd/lint.js

```diff
@@ -5,11 +5,59 @@ import {FileFilter} from './build';
 
 const log = createLogger(__filename);
 
+// Flow types
+
+export type LinterLogLevel = 'debug' | 'fatal';
+export type LinterOutputType = 'text' | 'json';
+
+export type LinterCreatorParams = {
+  config: {
+    logLevel: LinterLogLevel,
+    stack: boolean,
+    pretty?: boolean,
+    metadata?: boolean,
+    output?: LinterOutputType,
+    boring?: boolean,
+    selfHosted?: boolean,
+    shouldScanFile: (fileName: string) => boolean,
+    _: Array<string>,
+  },
+};
+
+export type Linter = {
+  run: () => Promise<void>,
+};
+
+export type LinterCreatorFn = (params: LinterCreatorParams) => Linter;
+
+
+export type LintCmdParams = {
+  sourceDir: string,
+  verbose?: boolean,
+  selfHosted?: boolean,
+  boring?: boolean,
+  output?: LinterOutputType,
+  metadata?: boolean,
+  pretty?: boolean,
+};
+
+export type LintCmdExtraParams = {
+  createLinter: LinterCreatorFn,
+  fileFilter: FileFilter,
+};
+
+// Module internals & exports
+
 export default function lint(
-    {verbose, sourceDir, selfHosted, boring, output,
-     metadata, pretty}: Object,
-    {createLinter=defaultLinterCreator, fileFilter=new FileFilter()}
-    : Object = {}): Promise<void> {
+  {
+    verbose, sourceDir, selfHosted, boring, output,
+    metadata, pretty,
+  }: LintCmdParams,
```

Comment:
yes, it is very unfortunate :-(

anyway, I noticed that "declaring the types of this destructured objects", besides reducing mistakes, forces us to re-look and re-think the type of this "options" parameters (something like the kind of additional thinking that the testing suite forces us to do on the code that we wrote), and I found it an useful exercise.